packages/core/test/oracle/VotingV2.js

```diff
@@ -3024,4 +3024,88 @@ describe("VotingV2", function () {
+    await supportedIdentifiers.methods.addSupportedIdentifier(identifier3).send({ from: accounts[0] });
+    await supportedIdentifiers.methods.addSupportedIdentifier(identifier4).send({ from: accounts[0] });
+
+    await voting.methods.requestPrice(identifier1, time1).send({ from: registeredContract });
+    await voting.methods.requestPrice(identifier2, time2).send({ from: registeredContract });
+    await voting.methods.requestPrice(identifier3, time3).send({ from: registeredContract });
+    await voting.methods.requestPrice(identifier4, time4).send({ from: registeredContract });
+
+    // Move to the voting round.
+    await moveToNextRound(voting, accounts[0]);
+    const roundId = (await voting.methods.getCurrentRoundId().call()).toString();
+
+    // Commit vote 1 and 4.
+    const price = getRandomSignedInt();
+    const salt = getRandomSignedInt();
+    const hash4 = computeVoteHash({
+      price: price,
+      salt: salt,
+      account: account1,
+      time: time4,
+      roundId,
+      identifier: identifier4,
+    });
+    const hash1 = computeVoteHash({
+      price: price,
+      salt: salt,
+      account: account1,
+      time: time1,
+      roundId,
+      identifier: identifier1,
+    });
+
+    await voting.methods.commitVote(identifier4, time4, hash4).send({ from: account1 });
+    await voting.methods.commitVote(identifier1, time1, hash1).send({ from: account1 });
+
+    // Move to the reveal phase of the voting period.
+    await moveToNextPhase(voting, account1);
+
+    // Reveal vote.
+    await voting.methods.revealVote(identifier4, time4, price, salt).send({ from: account1 });
+    await voting.methods.revealVote(identifier1, time1, price, salt).send({ from: account1 });
+
+    await moveToNextRound(voting, account1);
+
+    assert.isTrue(await voting.methods.hasPrice(identifier4, time4).call({ from: registeredContract }));
+    assert.isTrue(await voting.methods.hasPrice(identifier1, time1).call({ from: registeredContract }));
+    assert.isFalse(await voting.methods.hasPrice(identifier2, time2).call({ from: registeredContract }));
+    assert.isFalse(await voting.methods.hasPrice(identifier3, time3).call({ from: registeredContract }));
+
+    assert.equal(
+      (await voting.methods.getPrice(identifier4, time4).call({ from: registeredContract })).toString(),
+      price.toString()
+    );
+    assert.equal(
+      (await voting.methods.getPrice(identifier1, time1).call({ from: registeredContract })).toString(),
+      price.toString()
+    );
+
+    await voting.methods.updateTrackers(account1).send({ from: account1 });
+    await voting.methods.updateTrackers(account2).send({ from: account1 });
+
+    // Both users should have the same lastRequestIndexConsidered.
+    assert.equal((await voting.methods.voterStakes(account1).call()).lastRequestIndexConsidered, 4);
+    assert.equal((await voting.methods.voterStakes(account2).call()).lastRequestIndexConsidered, 4);
+  });
```

Comment:
How does this validate the new behavior? Did something above revert before? Or were these indices wrong before?

Reply:
Hey @mrice32, yes this last assertion failed before the changes:
`assert.equal((await voting.methods.voterStakes(account2).call()).lastRequestIndexConsidered, 4);`

Explanation:
The consecutive rolled votes were not handled properly. The second person processing the trackers with a second rolled vote would enter [this break](https://github.com/UMAprotocol/protocol/blob/0860b59dbf1ca437d45176d4585de74f7f034740/packages/core/contracts/oracle/implementation/VotingV2.sol#L867) and that was the end of it, leaving the next trackers unprocessed resulting in different `lastRequestIndexConsidered` across users.